src/app/backend/validation/validateloginstatus.go

```diff
@@ -32,13 +32,18 @@ type LoginStatus struct {
 }
 
 // ValidateLoginStatus returns information about user login status and if request was made over HTTPS.
-func ValidateLoginStatus(request *restful.Request) *LoginStatus {
+func ValidateLoginStatus(request *restful.Request, enableInsecureLogin bool) *LoginStatus {
 	authHeader := request.HeaderParameter("Authorization")
 	tokenHeader := request.HeaderParameter(client.JWETokenHeader)
 
+	httpsMode := request.Request.TLS != nil
```

Comment:
`httpsMode := request.Request.TLS != nil || enableInsecureLogin` if it is not too complex.